features/cellular/framework/AT/ATHandler.cpp

```diff
@@ -1099,7 +1097,12 @@ size_t ATHandler::write(const void *data, size_t len)
     fhs.events = POLLOUT;
     size_t write_len = 0;
     for (; write_len < len; ) {
-        int count = poll(&fhs, 1, _at_timeout);
+        int timeout = (_start_time + _at_timeout) - rtos::Kernel::get_ms_count();
```

Comment:
This should probably have the same timeout computation logic as for the fill_buffer, so attempting something if time has passed. So maybe make it a "compute timeout" subroutine.

You only need the device error on the return from poll, not this extra one.